api/src/db/migrations/v159_remove_entity_id.ts

```diff
@@ -0,0 +1,992 @@
+  "CORPORATION",
+  "INCORPORATED",
+  "CORP",
+  "CORP.",
+  "INC",
+  "INC.",
+] as const;
+
+const foreignCorpBusinessSuffix = [...corpBusinessSuffix, "P.C.", "P.A."] as const;
+
+export const AllBusinessSuffixes = [
+  ...llcBusinessSuffix,
+  ...llpBusinessSuffix,
+  ...lpBusinessSuffix,
+  ...corpBusinessSuffix,
+  ...foreignCorpBusinessSuffix,
+  ...nonprofitBusinessSuffix,
+] as const;
+
+type v159BusinessSuffix = (typeof AllBusinessSuffixes)[number];
+
+type v159FormationSubmitResponse = {
+  success: boolean;
+  token: string | undefined;
+  formationId: string | undefined;
+  redirect: string | undefined;
+  errors: v159FormationSubmitError[];
+  lastUpdatedISO: string | undefined;
+};
+
+type v159FormationSubmitError = {
+  field: string;
+  type: "FIELD" | "UNKNOWN" | "RESPONSE";
+  message: string;
+};
+
+type v159GetFilingResponse = {
+  success: boolean;
+  entityId: string;
+  transactionDate: string;
+  confirmationNumber: string;
+  formationDoc: string;
+  standingDoc: string;
+  certifiedDoc: string;
+};
+
+export type v159EnvironmentData = {
+  waste?: v159WasteData;
+  land?: v159LandData;
+  air?: v159AirData;
+};
+
+export type v159MediaArea = keyof v159EnvironmentData;
+export type v159QuestionnaireFieldIds =
+  | v159WasteQuestionnaireFieldIds
+  | v159LandQuestionnaireFieldIds
+  | v159AirQuestionnaireFieldIds;
+export type v159Questionnaire = Record<v159QuestionnaireFieldIds, boolean>;
+export type v159QuestionnaireConfig = Record<v159QuestionnaireFieldIds, string>;
+
+export type v159WasteData = {
+  questionnaireData: v159WasteQuestionnaireData;
+  submitted: boolean;
+};
+
+export type v159WasteQuestionnaireFieldIds =
+  | "hazardousMedicalWaste"
+  | "compostWaste"
+  | "treatProcessWaste"
+  | "constructionDebris"
+  | "noWaste";
+
+export type v159WasteQuestionnaireData = Record<v159WasteQuestionnaireFieldIds, boolean>;
+
+export type v159LandData = {
+  questionnaireData: v159LandQuestionnaireData;
+  submitted: boolean;
+};
+
+export type v159LandQuestionnaireFieldIds =
+  | "takeOverExistingBiz"
+  | "propertyAssessment"
+  | "constructionActivities"
+  | "siteImprovementWasteLands"
+  | "noLand";
+
+export type v159LandQuestionnaireData = Record<v159LandQuestionnaireFieldIds, boolean>;
+
+export type v159AirData = {
+  questionnaireData: v159AirQuestionnaireData;
+  submitted: boolean;
+};
+
+export type v159AirQuestionnaireFieldIds =
+  | "emitPollutants"
+  | "emitEmissions"
+  | "constructionActivities"
+  | "noAir";
+
+export type v159AirQuestionnaireData = Record<v159AirQuestionnaireFieldIds, boolean>;
+
+export type v159TaxClearanceCertificateData = {
+  requestingAgencyId?: string | undefined;
+  businessName: string | undefined;
+  addressLine1: string | undefined;
+  addressLine2: string | undefined;
+  addressCity: string | undefined;
+  addressState?: v159StateObject | undefined;
+  addressZipCode: string | undefined;
+  taxId: string | undefined;
+  taxPin: string | undefined;
+};
```

Comment:
[pebble] Given we're updating the type here in the migration file, we should also update the type definition used throughout the application in `taxClearanceCertificate.ts`.

When a new user signs up, we create an empty business record for them. That function `createEmptyBusiness` lives in `userData.ts`. On line 75 it sets `taxClearanceCertificateData` to `emptyTaxClearanceCertificateData`. We can also set some of the values in  `emptyTaxClearanceCertificateData` to `undefined`.

Reply:
The entity id field should also be removed from emptyTaxClearanceCertificateData.